core/services/wifi/Hotspot.py

```diff
@@ -0,0 +1,339 @@
+import pathlib
+import shlex
+import shutil
+import subprocess
+import tempfile
+import time
+from enum import Enum
+from ipaddress import IPv4Address
+from typing import Any, List, Optional
+
+import psutil
+from commonwealth.utils.DHCPServerManager import Dnsmasq as DHCPServerManager
+from loguru import logger
+from pyroute2 import IW, IPRoute
+
+
+class HostapdFrequency(str, Enum):
+    """Valid hostapd frequency modes."""
+
+    HW_2_4 = "g"  # Hostapd id for 2.4 GHz mode
+    HW_5_0 = "a"  # Hostapd id for 5.0 GHz mode
+
+    @staticmethod
+    def mode_from_channel_frequency(channel: int) -> "HostapdFrequency":
+        return HostapdFrequency.HW_2_4 if str(channel).startswith("2") else HostapdFrequency.HW_5_0
+
+
+class HotspotManager:
+    # pylint: disable=too-many-instance-attributes
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        base_interface: str,
+        ipv4_gateway: IPv4Address,
+        ap_interface_name: str = "uap0",
+        ap_ssid: Optional[str] = None,
+        ap_passphrase: str = "blueosap",
```

Comment:
It already is. We make the SSID `None` just because we calculate its value from a method (it's not a plain string). Will make both equal to improve readability though.